client/platform/desktop/frontend/store/index.ts

```diff
@@ -0,0 +1,54 @@
+import { Api, Pipe } from 'viame-web-common/apispec';
+import * as api from 'platform/desktop/frontend/api';
+
+/* Warning, this import involves node.js code for loadDetections (below) */
+import common from 'platform/desktop/backend/native/common';
+
+import { settings } from './settings';
+import { getRecents, setRecents, RecentsKey } from './dataset';
+import { getOrCreateHistory } from './jobs';
+
+/* Run forward migrations on any client-side data stores */
+export async function migrate() {
+  const recents = await getRecents();
+  if (recents.length && typeof recents[0] === 'string') {
+    window.localStorage.setItem(RecentsKey, JSON.stringify([]));
+  }
+}
+
+/**
+ * Wrap API with hooks to use the store
+ */
+export default function wrap(): Api {
+  async function runPipeline(itemId: string, pipeline: Pipe) {
+    const job = await api.runPipeline(itemId, pipeline);
+    getOrCreateHistory(job, job.datasetIds);
+  }
+
+  async function loadMetadata(datasetId: string) {
+    const meta = await api.loadMetadata(datasetId);
+    setRecents(meta);
+    return meta;
+  }
+
+  /**
+   * loadDetections loads JSON data directly from disk into the
+   * renderer thread. It relies on the node runtime being enabled on the browser.
+   *
+   * This is done such that large annotation files do not need to be loaded into memory
+   * twice, serialized and deserialized twice, and transmitted over the local network.
+   *
+   * In a future version, this could me moved to the backend and streamed directly
+   * to the client using something like https://github.com/uhop/stream-json
```

Comment:
interesting side-note, exact package I was looking at when wondering if there should be streaming for large json file conversions to CSV, but decided against it because all JSON files need to be fully loaded in memory anyways.

Reply:
I know, I got this from you.

I put the comment in just as a relevant bookmark since github issue / PR search is noisy. Can remove if you want.

Reply:
yeah leave it in as a reminder.  I'm assuming that at some point we are going to be hitting track files that may be 10's of MBs and transferring them all at once and waiting to load may not be feasible.